crates/miden-objects/src/block/header.rs

```diff
@@ -46,30 +46,21 @@ pub struct BlockHeader {
 
 impl BlockHeader {
     /// Creates a new block header.
-    #[allow(clippy::too_many_arguments)]
     pub fn new(
+        commitments: BlockHeaderCommitments,
```

Comment:
`BlockHeader::new` needs 12 arguments so IMO this is the only place worth of a refactoring.

All commitment-like elements were grouped into a common `BlockHeaderCommitments` structure that contains adhoc documentations. AFAICT no external API contract broke due to this change.

Reply:
I'm not sure if getting rid of the too many arguments lint is worth adding the extra struct.
If we really want the struct, we should also move account and nullifier root in there as well, but I don't really see a big advantage of it. It's not a very frequently used API and having to provide all parameters at once seems actually cleaner than having to wrap them in a separate struct. Personally, I would leave it the way it is now.

Reply:
Sure, no problem. All changes related to `BlockHeader` have been reverted.